src/function/cast_from_string_functions.cpp

```diff
@@ -305,16 +305,18 @@ static bool splitCStringList(const char* input, uint64_t len, T& state, const CS
     }
     skipWhitespace(++input, end);
 
+    bool just_finished_entry = true; // true at start
     auto start_ptr = input;
     while (input < end) {
         auto ch = *input;
         if (ch == CopyConstants::DEFAULT_CSV_LIST_BEGIN_CHAR) {
             if (!skipToClose(input, end, ++lvl, CopyConstants::DEFAULT_CSV_LIST_END_CHAR, option)) {
                 return false;
             }
-        } else if (ch == '\'' || ch == '"') {
+        } else if ((ch == '\'' || ch == '"') && just_finished_entry) {
```

Comment:
What's the duckdb's behaviour there?
If we don't care whether quotes are closed, we should simply remove the check there?
We allow users to customize the quote character. Should we check for that?

Reply:
Unfortunately DuckDB's grammar is not documented and it is a bit hard to reverse engineer. From what I can tell, it seems to treat both quotes the same, and implicitly adds quotes for entries that weren't surrounded by quotes (i.e. "[abc, xyz]" -> "['abc', 'xyz']"). I believe with this we would support a superset of DuckDB's grammar.

We need to have the check to skip the contents inside a quote so we don't try to parse it. For example, `['ab[', 'xyz']` should be valid - we don't want to try and parse the opening `[` in the first entry as another list.

I don't think the customizable quote character should apply to lists, e.g. I don't think we would want `[#abc#, #xyz#]` to be possible - DuckDB doesn't allow this customization either.